src/main/webapp/app/localci/build-queue/build-queue.component.ts

```diff
@@ -12,6 +12,88 @@ import { onError } from 'app/shared/util/global.utils';
 import { HttpErrorResponse, HttpHeaders, HttpResponse } from '@angular/common/http';
 import { AlertService } from 'app/core/util/alert.service';
 import dayjs from 'dayjs/esm';
+import { NgbModal, NgbTypeahead } from '@ng-bootstrap/ng-bootstrap';
+import { HttpParams } from '@angular/common/http';
+import { LocalStorageService } from 'ngx-webstorage';
+import { Observable, OperatorFunction, Subject, merge } from 'rxjs';
+import { debounceTime, distinctUntilChanged, map, switchMap, tap } from 'rxjs/operators';
+import { UI_RELOAD_TIME } from 'app/shared/constants/exercise-exam-constants';
+
+export class FinishedBuildJobFilter {
+    status?: string = undefined;
+    buildAgentAddress?: string = undefined;
+    buildStartDateFilterFrom?: dayjs.Dayjs = undefined;
+    buildStartDateFilterTo?: dayjs.Dayjs = undefined;
+    buildDurationFilterLowerBound?: number = undefined;
+    buildDurationFilterUpperBound?: number = undefined;
+
+    /**
+     * Adds the http param options
+     * @param options request options
+     */
+    addHttpParams(options: HttpParams): HttpParams {
+        if (this.status && this.status !== '') {
+            options = options.append('buildStatus', this.status);
+        }
+        if (this.buildAgentAddress && this.buildAgentAddress !== '') {
+            options = options.append('buildAgentAddress', this.buildAgentAddress);
+        }
+        if (this.buildStartDateFilterFrom) {
+            options = options.append('startDate', this.buildStartDateFilterFrom.toISOString());
+        }
+        if (this.buildStartDateFilterTo) {
+            options = options.append('endDate', this.buildStartDateFilterTo.toISOString());
+        }
+        if (this.buildDurationFilterLowerBound) {
+            options = options.append('buildDurationLower', this.buildDurationFilterLowerBound.toString());
+        }
+        if (this.buildDurationFilterUpperBound) {
+            options = options.append('buildDurationUpper', this.buildDurationFilterUpperBound.toString());
+        }
+
+        return options;
+    }
+
+    /**
+     * Returns the number of applied filters.
+     */
+    get numberOfAppliedFilters(): number {
+        return Object.values(this).filter((value) => value).length;
+    }
+
+    /**
+     * Checks if the dates are valid.
+     */
+    get areDatesValid(): boolean {
+        if (!this.buildStartDateFilterFrom || !this.buildStartDateFilterTo) {
+            return true;
+        }
+        return dayjs(this.buildStartDateFilterFrom).isBefore(dayjs(this.buildStartDateFilterTo));
+    }
+
+    get areDurationFiltersValid(): boolean {
+        if (!this.buildDurationFilterLowerBound || !this.buildDurationFilterUpperBound) {
+            return true;
+        }
+        return this.buildDurationFilterLowerBound < this.buildDurationFilterUpperBound;
+    }
```

Comment:
Be careful when accessing these values in the template - they don't look like function calls there, but they are. Maybe you can store these values as proper attributes in the filter and update them whenever the filter changes? That way, you wouldn't have to rely on it constantly being recomputed via the template.